### PR TITLE
Sujan

### PR DESCRIPTION
**Pull request by vsujan**
i have added a detail view to show the description of vaccine, datepicker for vaccine checked and notification alert.

### CODE REVIEW

File: Vaccination/VaccinationCell.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
+//
+//  VaccinationCell.swift
+//  Vaccination
+//
+//  Created by LF Technology on 2/29/16.
+//  Copyright © 2016 LFTechnology. All rights reserved.
+//
+
```
Comment (sanjaymhj) [inline Vaccination/VaccinationCell.swift]: Remove auto generated code at start of the file

File: Vaccination/VaccinationCell.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    @IBOutlet weak var dateLabel: UILabel!
+    @IBOutlet weak var monthLabel: UILabel!
+    @IBOutlet weak var vaccinationName: UILabel!
+    @IBOutlet weak var timeRemaining: UILabel!
+    
+    @IBOutlet var notificationImage: UIButton!
+    
+    @IBOutlet var checkedImage: UIButton!
+
```
Comment (sanjaymhj) [inline Vaccination/VaccinationCell.swift]: remove extra lines

File: Vaccination/VaccinationListTVC.swift [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import UIKit
+
+class VaccinationListTVC: UITableView {
+}
```
Comment (kusalshrestha) [inline Vaccination/VaccinationListTVC.swift]: Add  a new line at the end of the file

File: Vaccination/ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+            "date":"1",
+            "month":"Jan",
+            "vaccinationName":"BCG",
+            "timeRemaining":"2",
+            "vaccinationGiven":"true",
+            "vaccinationAlert":"true"
+        ]
+    ]
+    var addChildForm: Bool = false
+    var showGenderOption: Bool = false
```
Comment (kusalshrestha) [inline Vaccination/ViewController.swift]: You dont need to define Bool if you preinitialize variable with bool value

File: Vaccination/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+    @IBOutlet weak var coverImage: UIImageView!
+    @IBOutlet weak var mainScrollView: MainScrollView!
+    @IBOutlet weak var segmentController: UISegmentedControl!
+    @IBOutlet weak var vaccinationListTVC: VaccinationListTVC!
+    @IBOutlet var formHeightConstraint: NSLayoutConstraint!
+    @IBOutlet var month: UITextField!
+    @IBOutlet var day: UITextField!
+    @IBOutlet var year: UITextField!
+    @IBOutlet var genderTVC: GenderTVC!
+    @IBOutlet var gender: UILabel!
```
Comment (kusalshrestha) [inline Vaccination/ViewController.swift]: USe proper naming for the outlets.. "month" -> "monthTextField"...

File: Vaccination/ViewController.swift [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.vaccinationListTVC.delegate = self
+        self.vaccinationListTVC.dataSource = self
+        self.genderTVC.delegate = self
+        self.genderTVC.dataSource = self
+        self.mainScrollView.delegate = self
+        self.formHeightConstraint.constant = 0
+        genderTVC.hidden = true
```
Comment (kusalshrestha) [inline Vaccination/ViewController.swift]: Better if you assign delegate and data source from the storyboard as well

File: Vaccination/ViewController.swift [hunk 7]
```diff
@@ -0,0 +1,448 @@
...
+        super.viewWillAppear(true)
+        self.vaccinationListTVC.scrollEnabled = false
+        self.vaccinationListTVC.scrollsToTop = false
+//        self.form.hidden = true
+    }
+
+    @IBAction func segmentSelected(sender: AnyObject) {
+        if segmentController.selectedSegmentIndex == 0 {
+            tableData = dataUpcoming
+        } else if segmentController.selectedSegmentIndex == 1 {
```
Comment (kusalshrestha) [inline Vaccination/ViewController.swift]: use enum for the segmentIndex

File: Vaccination/ViewController.swift [hunk 8]
```diff
@@ -0,0 +1,448 @@
...
+            tableData = dataHistory
+        }
+        self.selectedIndexPath = nil
+        self.vaccinationListTVC.reloadData()
+    }
+    
+    @IBAction func selectGender(sender: AnyObject) {
```
Comment (kusalshrestha) [inline Vaccination/ViewController.swift]: You can select the sender type while making the action from the IB

File: Vaccination/ViewController.swift [hunk 9]
```diff
@@ -0,0 +1,448 @@
...
+                    selectedIndexPath = indexPath
+                }
+            }
+            tableView.reloadRowsAtIndexPaths([indexPath], withRowAnimation: UITableViewRowAnimation.Fade)
+        }
+    }
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let smallHeight: CGFloat = 114.0
+        let expandedHeight: CGFloat = 270.0
+        let ip = indexPath
```
Comment (kusalshrestha) [inline Vaccination/ViewController.swift]: use a proper naming for the variables

File: Vaccination/ViewController.swift [hunk 10]
```diff
@@ -0,0 +1,448 @@
...
+        } else {
+            return smallHeight
+        }
+    }
+    
+}
+
+extension ViewController: UITableViewDataSource {
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
```
Comment (kusalshrestha) [inline Vaccination/ViewController.swift]: count for the section is default to 1. You dont need to write this code for single section

File: Vaccination/VaccinationListTVC.swift (hunk 3 above)
Comment (vsujan) [inline Vaccination/VaccinationListTVC.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 8 above)
Comment (vsujan) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 9 above)
Comment (vsujan) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 10 above)
Comment (vsujan) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/VaccinationCell.swift (hunk 1 above)
Comment (vsujan) [inline Vaccination/VaccinationCell.swift]: :ok:

File: Vaccination/VaccinationCell.swift (hunk 2 above)
Comment (vsujan) [inline Vaccination/VaccinationCell.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 4 above)
Comment (vsujan) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/CoreData/CoreDataHelper.swift [hunk 11]
```diff
@@ -88,7 +88,7 @@ class CoreDataHelper: CoreDataStack {
      - parameter dictionary: NSDictionary
      */
     func insertEntity(dictionary: NSDictionary) {
-        if let entityID = Int(dictionary.objectForKey("id") as! String) {
+        if let entityID = Int((dictionary.objectForKey("id") as? String)!) {
```
Comment (leapfrog-saugat) [inline Vaccination/CoreData/CoreDataHelper.swift]: There should be two levels of optional unwrapping in this case. Once for checking nil value in dictionary and another for checking if string value was successfully cast into Int.

**Pull request closed without merging**